### ISSUE TITLE
raspbian stretch, PHP7 ssh2 and public key

### ISSUE THREAD

**Issue by TT-SWP13** [gumslone/GumCP#12]
Hello,

i installed GumCP on a RPi3 with Raspbian Stretch. To get _php5-ssh2_ i edit /etc/apt/sources.list from _stretch_ to _jessie_.

1. sudo nano /etc/apt/sources.list
- change line "deb http://mirrordirector.raspbian.org/raspbian/ _stretch_ main contrib non-free rpi" to "deb http://mirrordirector.raspbian.org/raspbian/ _jessie_ main contrib non-free rpi"
2. sudo apt-get update
3. sudo apt-get install php5-ssh2
4. undo change in sources.list
5. /etc/init.d/php5-fpm restart
6. sudo service nginx restart
Now it works!

But here my two questions now:
**It is possible to use php7?**
**It is possible to use a ssh public-key?**

http://php.net/manual/de/function.ssh2-auth-pubkey-file.php 

Thank you and best regards!

**Comment by gumslone** [comment]
with php7 it should work too, you just need to install a different ssh2 library for php7, I have one raspberry pi zero w at home with php 7 and GumCP running well.
For PHP 7:

`sudo apt-get install php7.0-ssh2`
For PHP 7.1:

`sudo apt-get install php7.1-ssh2`
Restart Apache2 server
`sudo service apache2 restart`

php ssh is required to send commands as sudo user

The public key should work with it too, i haven't tested it yet.

**Comment by nigelhaslam** [comment]
Hi Gumslone,

I'm loving your GumCP interface which I got up and running in 10 minutes without a hitch - needed the information above from TT-SWP13, to change source repositories though.  Might be good to update the main readme to reflect that need.

After installing php5-ssh, I checked my **/etc/php5/apache2/php.ini** file and found it to contain the text **extension=ssh**. can I therefore assume that my connection to my Raspberry Pi is using ssh and therefore secure?  I'm not very tech savvy but in an effort to secure the web connection to my RaspPi (currently home-office based),  I've forwarded a non standard port to the Pi's LAN address (internal port 22) 

Particularly useful was the background command output log, which enabled me to debug the script I was trying to invoke but which wasn't working.

Cheers
Nigel
Byron Bay
Australia

**Comment by gumslone** [comment]
Hi Nigel, 
if your pi is accessible from internet, outside of local network, its strongly recommended not to use default ssh password for user pi.
To check if php-ssh2 is installed correctly, you could try to open the Actions page, if it opens without errors then its installed.
Ssh on port 22 of your pi and php-ssh2 are different things, php-ssh2 is a library which makes it possible for php script to connect to ssh port 22 of your pi and send commands as superuser "sudo", (therefor you have to provide your ssh credentials in the config file), php-ssh can be also used for connection to external servers too.

**Comment by nigelhaslam** [comment]
Thanks mate,

Aside from using a non standard external port, forwarded to the Pi which is
currently networked on my office LAN, I changed the ssh password in the
config.

Thanks for the insight into the differences between ssh on the pi and
php-ssh.

I struggled a little at first and thought it was ssh related but it turned
out to be the lack of a full path to a dependency in the script.  When I
ran my command as a background process, I got a text file which clarified
the error, so I fixed up the path and now it's all working nicely.

Thanks so much for sharing this.  I must say I thought it was going to be a
long climb up a steep learning curve to achieve my needs but you've given
me a lift up to the top!

All I need now is to integrate some admin tools and a schedule timer for my
time lapse scripts ie.

Admin

   - Set image name and resolution
   - Set thumbnail resolution
   - Manage stored images - download / delete
   - Image capture interval

Scheduler

   - flexible task control
   - Could add cron entries

Cheers

*Nigel Haslam* *| Director *
